### ISSUE TITLE
Wax accessiblity fixes for HHMI

### ISSUE THREAD

**Issue by yannisbarlas** [Coko-Foundation/wax-prosemirror#69]
> **Migrated from GitLab**
> **Original Author:** Grigor Malo (`@grgml`)
> **Original Link:** https://gitlab.coko.foundation/wax/wax-prosemirror/-/issues/57

Opening this issue to keep track of pending changes for wax. These are accessibility improvements requested by HHMI

1. [ ] Multiple choice or true/false questions: Programmatic label does not convey purpose of control  
The label of the switch button (yes/no or true/false) is not descriptive / does not give context about what the current option is  
Current label: Is it correct? (checked/unchecked)  
Improved label: Is this correct? (read question) (checked/unchecked)  

2. [ ] multiple choice or true/false questions: Buttons have same name but different actions  
The problem is about the add/delete buttons.
"Multiple buttons have the same labels/names but perform different actions. This should be conveyed via text associated with individual     buttons. Here multiple buttons are present for different questions.Make sure to provide unique programmatic label for the buttons."
Examples of improved labels: "Add new option below this one", "Delete option: (read option text)"

3. [ ] Matching type questions  
  - Function cannot be performed by keyboard alone  
  - Listbox: Listbox is missing appropriate roles and/or attributes (in test mode)

4. ~Toolbar Button: Button pressed state is not conveyed  
Not sure why, but buttons in the toolbar cannot be toggled with Enter or Space keys
"aria-press attribute is provided but it is not actionable with keyboard hence screen reader is not announcing its press state."~

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Dione Mentis (`@DioneMentis`)
> **Date:** 2023-05-18T12:07:36.789Z

`@christos`  for point 1 on multiple choice questions "Programmatic label does not convey purpose of control" I think it would be better to remove the `Correct ?` text in the UI and rename the toggles to be accurate `Correct` and `Incorrect`

![image](/uploads/028ef4c2225132bcc38bc24d204add96/image.png)

If we apply the same logic to T/F questions, then we also resolve the issue in https://gitlab.coko.foundation/hhmi/hhmi/-/issues/107 because the user can provide "True" or "False" as the answer text, then toggle  `Correct` and `Incorrect`. 

What do you think? 

`@grgml`  Do you agree?

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Grigor Malo (`@grgml`)
> **Date:** 2023-05-18T14:19:00.991Z

That wouldn't completely solve the problem. I think the complaint is about not having a programmatic relationship between the options' text and the switch control. So when focus lands on the switch, the screen reader should announce the option we are assessing as correct or incorrect.

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Christos (`@christos`)
> **Date:** 2023-05-18T17:53:27.941Z

`@DioneMentis`  no worries i have already done that . the screen reader will announce "is it correct + the text that was entered in the answer editor part"

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Dione Mentis (`@DioneMentis`)
> **Date:** 2023-05-23T11:00:05.027Z

`@christos`  thanks. I'll move my suggested here to an "design improvement" issue in the HHMI repo

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Dione Mentis (`@DioneMentis`)
> **Date:** 2023-05-23T11:01:52.466Z

`@christos`  are all these updates published in a new version of wax?

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Christos (`@christos`)
> **Date:** 2023-05-23T11:13:10.247Z

Most likely on Thursday.

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Dione Mentis (`@DioneMentis`)
> **Date:** 2023-05-29T07:54:13.175Z

`@christos`  do you have an update on this?

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Dione Mentis (`@DioneMentis`)
> **Date:** 2023-06-19T10:00:12.358Z

I've removed point 4 because it's not a feasible change. Since Wax in a editor, the focus must remain in the editor when a button in the toolbar is clicked, otherwise the user would need to take two actions whenever selecting a button in the toolbar (click button, click back in editor).

**Comment by yannisbarlas** [comment]
> **Migrated Comment**
> **Author:** Christos (`@christos`)
> **Date:** 2023-07-03T08:52:51.330Z

done in https://gitlab.coko.foundation/wax/wax-prosemirror/-/merge_requests/482. Also replaced the dropdown in multiple dropdown questions as there was the same issue with matching.
